### PR TITLE
[web] Support platform view creation params

### PR DESCRIPTION
**Pull request by mdebbar**
1. Wire platform views' `creation params` in the engine.
2. Move acceptable factory signatures to `dart:ui_web`.
3. Don't put any JS types in `dart:ui_web`'s interface.
    -  Use type casting to check at runtime instead.
4. Tests.

Part of https://github.com/flutter/flutter/issues/127030

### CODE REVIEW

File: lib/web_ui/test/engine/platform_views/message_handler_test.dart [hunk 1]
```diff
@@ -109,6 +111,64 @@ void testMain() {
...
+              isNull,
+              reason: 'The response should be a success envelope, with null in it.',
+            );
+          }
+
+          expect(factoryCalls, hasLength(2));
+          expect(factoryCalls[0].viewId, 111);
+          expect(factoryCalls[0].params, isNull);
+          expect(factoryCalls[1].viewId, 222);
+          expect(factoryCalls[1].params, <dynamic, dynamic>{'foo': 'bar'});
```
Comment (ditman) [inline lib/web_ui/test/engine/platform_views/message_handler_test.dart:155]: Do we need an extra case where the param passed is not a map, but a string or an int, or some other simple `Object`?

**Review by ditman — APPROVED** [review]
Let's go creation params!

File: lib/web_ui/lib/src/engine/platform_views/content_manager.dart [hunk 2]
```diff
@@ -121,10 +110,11 @@ class PlatformViewManager {
       final Function factoryFunction = _factories[viewType]!;
       late DomElement content;
 
-      if (factoryFunction is ParameterizedPlatformViewFactory) {
-        content = factoryFunction(viewId, params: params);
+      if (factoryFunction is ui_web.ParameterizedPlatformViewFactory) {
+        content = factoryFunction(viewId, params: params) as DomElement;
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/platform_views/content_manager.dart:114]: Very good idea!

[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/platform_views/content_manager.dart:114]: Actually, I don't think `DomElement` is a real type at runtime, so I'm not sure what `content is DomElement` is going to check. cc @joshualitt any ideas?

[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/platform_views/content_manager.dart:119]: I'm going to submit this as is. I can do a follow up PR if you still feel strongly about it.

**Pull request merged**